src/main/java/minkostplan/application/UIcontroller/HomepageController.java

```diff
@@ -29,7 +34,18 @@ public String admin(){
 
 
     @GetMapping("/profile")
-    public String profilepage(){
+    public String profilepage(Authentication authentication, Model model){
+        Users user = UserUtil.getCurrentUser();
+        System.out.println("User" + user);
+        model.addAttribute("User",user);
+        // Your POST method logic here
```

Comment:
Make comment into human and not AI talk
```suggestion
        //TODO make method logic for this.
```